src/pages/AddressInfoPage/index.tsx

```diff
@@ -59,118 +57,57 @@ type ParamTypes = {
   const isAppVisible = usePageVisibility()
   const pageNumber = usePageNumber()
+  const { displaySnackbar } = useSnackbar()
 
-  const [addressBalance, setAddressBalance] = useState<AddressBalance>()
-  const [addressTransactionNumber, setAddressTransactionNumber] = useState<number>()
-  const [addressAssets, setAddressAssets] = useState<AddressAssetsResult>()
-  const [addressTransactions, setAddressTransactions] = useState<Transaction[]>()
-  const [addressMempoolTransactions, setAddressMempoolTransactions] = useState<MempoolTransaction[]>([])
-  const [addressLatestActivity, setAddressLatestActivity] = useState<number>()
   const [addressWorth, setAddressWorth] = useState<number | undefined>(undefined)
   const [exportModalShown, setExportModalShown] = useState(false)
 
-  const [loadings, setLoadings] = useState<{ [key in AddressPropertyName]: boolean }>({
-    balance: true,
-    txNumber: true,
-    assets: true,
-    transactions: true
-  })
-
-  const fetchAddressDataGeneric = useCallback(
-    async <T,>(
-      addressPropName: AddressPropertyName,
-      setAddressProp: (value: T | undefined) => void,
-      apiCall: SingleStringArgFunctions<ExplorerProvider['addresses']> | (() => Promise<T>)
-    ) => {
-      if (!client || !id) return
-
-      setLoadings((p) => ({ ...p, [addressPropName]: true }))
-      setAddressProp(undefined)
-
-      try {
-        const result = typeof apiCall === 'string' ? ((await client.addresses[apiCall](id, {})) as T) : await apiCall()
-        setAddressProp(result)
-      } catch (error) {
-        displayError(setSnackbarMessage, error, `Error while fetching ${addressPropName}`)
-      } finally {
-        setLoadings((p) => ({ ...p, [addressPropName]: false }))
-      }
-    },
-    [client, id, setSnackbarMessage]
-  )
+  const lastKnownMempoolTxs = useRef<MempoolTransaction[]>([])
 
-  const fetchTransactions = useCallback(
-    () =>
-      client &&
-      id &&
-      fetchAddressDataGeneric('transactions', setAddressTransactions, async () => {
-        const currentPageTransactionData = await client.addresses.getAddressesAddressTransactions(id, {
-          page: pageNumber
-        })
-        const firstPageTransactionData = await client.addresses.getAddressesAddressTransactions(id, { page: 1 })
-        setAddressLatestActivity(firstPageTransactionData[0]?.timestamp)
-
-        return currentPageTransactionData
-      }),
-    [client, fetchAddressDataGeneric, id, pageNumber]
-  )
-
-  const fetchMempoolTxs = useCallback(
-    async (shouldTriggerTxFetch = true) => {
-      if (!client || !id) return
-
-      try {
-        const addressMempoolTransactionsHashes = new Set(addressMempoolTransactions.map((t) => t.hash))
-        const mempoolTxs = await client.addresses.getAddressesAddressMempoolTransactions(id)
+  const { data: addressBalance } = useQuery({
+    ...addressQueries.balance.details(addressHash),
+    enabled: !!addressHash
```

Comment:
I'm displaying an error is the addressHash isn't correct. The display could be improved in a following iteration though (remove the skeleton loading elements etc.)

Reply:
Could we add a TODO so that this comment doesn't get forgotten?

Reply:
Wouldn't it be useful to add the address validity check in the `enabled` flags of each query so that we avoid spamming the server with unnecessary requests when the address is invalid? Or is it too much for now?

Reply:
Ok. Let's merge this beast and I'll create a new PR for further improvements. WDYT?